code/datums/components/nanites.dm

```diff
@@ -317,6 +336,19 @@
 			mob_program["deactivation_code"] = P.deactivation_code
 			mob_program["kill_code"] = P.kill_code
 			mob_program["trigger_code"] = P.trigger_code
+			var/list/rules = list()
+			var/rule_id = 1
+			for(var/Z in P.rules)
+				var/datum/nanite_rule/nanite_rule = Z
+				var/list/rule = list()
+				rule["display"] = nanite_rule.display()
+				rule["program_id"] = id
+				rule["id"] = rule_id
+				rules += list(rule)
+				rule_id++

```

Comment:
this block is copypasted a few times

Reply:
once, which is because the cloud controller and the chamber computer have a few differences that prevent me from making an unified UI proc

Reply:
fair